### DISCUSSION TITLE
how its works in 2024

### DISCUSSION THREAD

**Discussion by Frankymuc** [IATkachenko/HA-SleepAsAndroid#69] [Q&A]
hallo , is this working in 2024?
i have the mosquito addon in home asistant. have installed the hacs integration sleep as android and start after boot HA the aplication. In my android samsung fold5 device i add the mqtt.
but i can do what i want...nothing works...also no complett example is found, no conret examples to configure. all test the app say not conected to mqtt.
what must i do:

tcp://mqtt_user:mqtt_password@mqtt_host:mqtt_port
my is:
tcp://f089:12345@192.168.2.230:1883
testbuton: connect fail

In the App:
Mqtt enabled
Url : tcp://f089:12345@192.168.2.230:1883
Thema: SleepAsAndroid
Client-ID: SleepAsAndroid
Intent Api : enabled

what mus in this string insert by device? the name from my device? Fold5 or what?
SleepAsAndroid/%%%device%%%

i dont understand this is so complicate witout step by step manual, all is only simplest and not for beginers..
where i found the host? i have only an IP from mosquito with port 1883..

**Comment by IATkachenko** [comment]
@Frankymuc, hello.

It works well (for me, at 2024.7.2).

If HA works well, but Application could not connect, then you have troubles with MQTT server setup. Connect string looks fine.
Are you using Mosquito as stand-alone server or HA-OS addon? Try to check Mosquito logs and increase log level.
(Personally, I'm using EMQX, because Mosquito have uncomfortable default.)
Try to use some MQTT guides for HomeAssistant if you have troubles at this stage. App or the integration have no special here: just ordinary read/write data from/to topic.

== Then the integration and Topic settings.

In the Application you must use some fixed topic, unique for every device. (`SleepAsAndroid/Igor` in my case). The application will write data to this topic. All different applications must have different topics.
In HA you MAY use static topic (same as in the App). In this case the integration will create just one device. Or you may use template (`SleepAsAndroid/%%%device%%%` in my case), then the integration will create instance for every found subtopic (`SleepAsAndroid/Igor` and `SleepAsAndroid/Mary` will create two independent devices in HA).

**Reply by Frankymuc** [reply]
thans for your answer..i used the mosquitto as HA-OS addon...the emqx was runing fine and month mai this year the troble beginn. all devices conected fine to emqx but only the zigbee2mqtt is not conectec stable..i can see its conected for 3 or 4 sec and go offline and this procedure all over the time on/off on/off..i dont know what the issue fron working to nor working.im sure i dont have chaned any setings or username or other..

last wee i have test with a very old backup to restore..its works perfect..withe the new backup and deinstall mosquito addon no working anymore..im back ro mosquitto and this works fine..but i like the emqs because many features and verry good statistics, this i like.

i have test any thinks but after mai this year - no chance...
